### ISSUE TITLE
Error installing app dubioushttp

### ISSUE THREAD

**Issue by noraj** [noxxi/p5-app-dubioushttp#4]
Hello,

I'm using perl stuff for the first time.

As far as I understand a perl module is currently installed like this:

```
$ perl Makefile.PL
$ make install
$ make test
# make install
```

or by doing `cpan App::DubiousHTTP`.

I have a warning when building:

```
$ perl Makefile.PL
Warning: prerequisite Compress::Raw::Lzma 2.036 not found.
Generating a Unix-style Makefile
Writing Makefile for App::DubiousHTTP
Writing MYMETA.yml and MYMETA.json
```

Changing `'Compress::Raw::Lzma' => 2.036,` in `Makefile.PL` to `'Compress::Raw::Lzma' => 2.082,` or `'Compress::Raw::Lzma' => 0,` change nothing.

But the module exists on CPAN and metacpan:

- http://search.cpan.org/~pmqs/Compress-Raw-Lzma-2.082/
- https://metacpan.org/pod/Compress::Raw::Lzma

So I tried this too:

```
$ cpan App::DubiousHTTP
Loading internal null logger. Install Log::Log4perl for logging messages

CPAN.pm requires configuration, but most of it can be done automatically.
If you answer 'no' below, you will enter an interactive dialog for each
configuration option instead.

Would you like to configure as much as possible automatically? [yes] yes
Use of uninitialized value $what in concatenation (.) or string at /usr/share/perl/5.24/App/Cpan.pm line 644, <STDIN> line 1.

Warning: You do not have write permission for Perl library directories.

To install modules, you need to configure a local Perl library directory or
escalate your privileges.  CPAN can help you by bootstrapping the local::lib
module or by configuring itself to use 'sudo' (if available).  You may also
resolve this problem manually if you need to customize your setup.

What approach do you want?  (Choose 'local::lib', 'sudo' or 'manual')
 [local::lib] sudo
Fetching with LWP:
http://www.cpan.org/authors/01mailrc.txt.gz
Reading '/home/noraj/.cpan/sources/authors/01mailrc.txt.gz'
............................................................................DONE
Fetching with LWP:
http://www.cpan.org/modules/02packages.details.txt.gz
Reading '/home/noraj/.cpan/sources/modules/02packages.details.txt.gz'
  Database was generated on Wed, 16 May 2018 13:17:02 GMT
.............
  New CPAN.pm version (v2.16) available.
  [Currently running version is v2.1101]
  You might want to try
    install CPAN
    reload cpan
  to both upgrade CPAN.pm and run the new version without leaving
  the current session.


...............................................................DONE
Fetching with LWP:
http://www.cpan.org/modules/03modlist.data.gz
Reading '/home/noraj/.cpan/sources/modules/03modlist.data.gz'
DONE
Writing /home/noraj/.cpan/Metadata
Running install for module 'App::DubiousHTTP'
Fetching with LWP:
http://www.cpan.org/authors/id/S/SU/SULLR/App-DubiousHTTP-0.033.tar.gz
Fetching with LWP:
http://www.cpan.org/authors/id/S/SU/SULLR/CHECKSUMS
Checksum for /home/noraj/.cpan/sources/authors/id/S/SU/SULLR/App-DubiousHTTP-0.033.tar.gz ok
'YAML' not installed, will not store persistent state
Configuring S/SU/SULLR/App-DubiousHTTP-0.033.tar.gz with Makefile.PL
Checking if your kit is complete...
Looks good
Warning: prerequisite Compress::Raw::Lzma 2.036 not found.
Generating a Unix-style Makefile
Writing Makefile for App::DubiousHTTP
Writing MYMETA.yml and MYMETA.json
  SULLR/App-DubiousHTTP-0.033.tar.gz
  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK
Running make for S/SU/SULLR/App-DubiousHTTP-0.033.tar.gz
---- Unsatisfied dependencies detected during ----
----    SULLR/App-DubiousHTTP-0.033.tar.gz    ----
    Compress::Raw::Lzma [requires]
Running install for module 'Compress::Raw::Lzma'
Fetching with LWP:
http://www.cpan.org/authors/id/P/PM/PMQS/Compress-Raw-Lzma-2.082.tar.gz
Fetching with LWP:
http://www.cpan.org/authors/id/P/PM/PMQS/CHECKSUMS
Checksum for /home/noraj/.cpan/sources/authors/id/P/PM/PMQS/Compress-Raw-Lzma-2.082.tar.gz ok
Configuring P/PM/PMQS/Compress-Raw-Lzma-2.082.tar.gz with Makefile.PL
Parsing config.in...
Looks Good.
Up/Downgrade not needed.
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Compress::Raw::Lzma
Writing MYMETA.yml and MYMETA.json
  PMQS/Compress-Raw-Lzma-2.082.tar.gz
  /usr/bin/perl Makefile.PL INSTALLDIRS=site -- OK
Running make for P/PM/PMQS/Compress-Raw-Lzma-2.082.tar.gz
cp lib/Compress/Raw/Lzma.pm blib/lib/Compress/Raw/Lzma.pm
AutoSplitting blib/lib/Compress/Raw/Lzma.pm (blib/lib/auto/Compress/Raw/Lzma)
Running Mkbootstrap for Compress::Raw::Lzma ()
chmod 644 "Lzma.bs"
"/usr/bin/perl" "/usr/share/perl/5.24/ExtUtils/xsubpp"  -typemap "/usr/share/perl/5.24/ExtUtils/typemap" -typemap "typemap"  Lzma.xs > Lzma.xsc && mv Lzma.xsc Lzma.c
x86_64-linux-gnu-gcc -c  -I/usr/local/include -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g   -DVERSION=\"2.082\" -DXS_VERSION=\"2.082\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.24/CORE"  -Wall -Wno-comment   Lzma.c
Lzma.xs:18:19: fatal error: lzma.h: No such file or directory
 #include "lzma.h"
                   ^
compilation terminated.
Makefile:347: recipe for target 'Lzma.o' failed
make: *** [Lzma.o] Error 1
  PMQS/Compress-Raw-Lzma-2.082.tar.gz
  /usr/bin/make -- NOT OK
  SULLR/App-DubiousHTTP-0.033.tar.gz
  Has already been unwrapped into directory /home/noraj/.cpan/build/App-DubiousHTTP-0.033-Itk2Gr
  SULLR/App-DubiousHTTP-0.033.tar.gz
  Has already been prepared
Running make for S/SU/SULLR/App-DubiousHTTP-0.033.tar.gz
Warning: Prerequisite 'Compress::Raw::Lzma => 2.036' for 'SULLR/App-DubiousHTTP-0.033.tar.gz' failed when processing 'PMQS/Compress-Raw-Lzma-2.082.tar.gz' with 'make => NO'. Continuing, but chances to succeed are limited.
cp lib/App/DubiousHTTP/Tests/Chunked.pm blib/lib/App/DubiousHTTP/Tests/Chunked.pm
cp lib/App/DubiousHTTP/TestServer.pm blib/lib/App/DubiousHTTP/TestServer.pm
cp lib/App/DubiousHTTP/Tests/Range.pm blib/lib/App/DubiousHTTP/Tests/Range.pm
cp lib/App/DubiousHTTP/Tests/Common.pm blib/lib/App/DubiousHTTP/Tests/Common.pm
cp lib/App/DubiousHTTP/Tests.pm blib/lib/App/DubiousHTTP/Tests.pm
cp lib/App/DubiousHTTP/Tests/MessageRfc822.pm blib/lib/App/DubiousHTTP/Tests/MessageRfc822.pm
cp lib/App/DubiousHTTP/Tests/Clen.pm blib/lib/App/DubiousHTTP/Tests/Clen.pm
cp lib/App/DubiousHTTP/Tests/Mime.pm blib/lib/App/DubiousHTTP/Tests/Mime.pm
cp lib/App/DubiousHTTP/Tests/Broken.pm blib/lib/App/DubiousHTTP/Tests/Broken.pm
cp lib/App/DubiousHTTP/Tests/TestID.pm blib/lib/App/DubiousHTTP/Tests/TestID.pm
cp lib/App/DubiousHTTP.pm blib/lib/App/DubiousHTTP.pm
cp lib/App/DubiousHTTP/Tests/Compressed.pm blib/lib/App/DubiousHTTP/Tests/Compressed.pm
cp bin/dubious_http.pl blib/script/dubious_http.pl
"/usr/bin/perl" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/dubious_http.pl
Manifying 1 pod document
  SULLR/App-DubiousHTTP-0.033.tar.gz
  /usr/bin/make -- OK
Running make test
PERL_DL_NONLAZY=1 PERL_USE_UNSAFE_INC=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/01_load.t ..... cannot load App::DubiousHTTP::Tests::Compressed: Can't locate Compress/Raw/Lzma.pm in @INC (you may need to install the Compress::Raw::Lzma module) (@INC contains: /home/noraj/.cpan/build/App-DubiousHTTP-0.033-Itk2Gr/blib/arch /home/noraj/.cpan/build/App-DubiousHTTP-0.033-Itk2Gr/blib/lib /home/noraj/.cpan/build/App-DubiousHTTP-0.033-Itk2Gr/blib/lib /home/noraj/.cpan/build/App-DubiousHTTP-0.033-Itk2Gr/blib/arch /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.24.1 /usr/local/share/perl/5.24.1 /usr/lib/x86_64-linux-gnu/perl5/5.24 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.24 /usr/share/perl/5.24 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base .) at /home/noraj/.cpan/build/App-DubiousHTTP-0.033-Itk2Gr/blib/lib/App/DubiousHTTP/Tests/Compressed.pm line 6.
BEGIN failed--compilation aborted at /home/noraj/.cpan/build/App-DubiousHTTP-0.033-Itk2Gr/blib/lib/App/DubiousHTTP/Tests/Compressed.pm line 6.
Compilation failed in require at (eval 6) line 2.
Compilation failed in require at bin/dubious_http.pl line 5.
BEGIN failed--compilation aborted at bin/dubious_http.pl line 5.
t/01_load.t ..... 1/1 
#   Failed test 'syntax check dubious_http.pl'
#   at t/01_load.t line 4.
# Looks like you failed 1 test of 1.
t/01_load.t ..... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
t/02_mapping.t .. cannot load App::DubiousHTTP::Tests::Compressed: Can't locate Compress/Raw/Lzma.pm in @INC (you may need to install the Compress::Raw::Lzma module) (@INC contains: /home/noraj/.cpan/build/App-DubiousHTTP-0.033-Itk2Gr/blib/lib /home/noraj/.cpan/build/App-DubiousHTTP-0.033-Itk2Gr/blib/arch /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.24.1 /usr/local/share/perl/5.24.1 /usr/lib/x86_64-linux-gnu/perl5/5.24 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.24 /usr/share/perl/5.24 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base .) at /home/noraj/.cpan/build/App-DubiousHTTP-0.033-Itk2Gr/blib/lib/App/DubiousHTTP/Tests/Compressed.pm line 6.
BEGIN failed--compilation aborted at /home/noraj/.cpan/build/App-DubiousHTTP-0.033-Itk2Gr/blib/lib/App/DubiousHTTP/Tests/Compressed.pm line 6.
Compilation failed in require at (eval 6) line 2.
Compilation failed in require at t/02_mapping.t line 3.
BEGIN failed--compilation aborted at t/02_mapping.t line 3.
t/02_mapping.t .. Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 

Test Summary Report
-------------------
t/01_load.t   (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
t/02_mapping.t (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output
Files=2, Tests=1,  1 wallclock secs ( 0.03 usr  0.00 sys +  0.10 cusr  0.02 csys =  0.15 CPU)
Result: FAIL
Failed 2/2 test programs. 1/1 subtests failed.
Makefile:886: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 2
  SULLR/App-DubiousHTTP-0.033.tar.gz
one dependency not OK (Compress::Raw::Lzma); additionally test harness failed
  /usr/bin/make test -- NOT OK
//hint// to see the cpan-testers results for installing this module, try:
  reports SULLR/App-DubiousHTTP-0.033.tar.gz
```

The CPAN module seams to be in version 0.033 and the git in version 0.034.

I have perl5 (revision 5 version 24 subversion 1) and Debian GNU/Linux 9 (stretch).

**Comment by noraj** [comment]
PS : dependency check with CPAN tester: http://deps.cpantesters.org/?module=App%3A%3ADubiousHTTP&perl=5.24.1&os=Linux

**Comment by noxxi** [comment]
Using `cpan` for installation is the correct way. But you have to look at the output where you will find that it fails to install Compress::Raw::Lzma since it obviously has no development files for the lzma library. See the [README for Compress::Raw::Lzma](https://metacpan.org/source/PMQS/Compress-Raw-Lzma-2.082/README) for details on where you can find this library.

**Comment by noraj** [comment]
I have those debian packages installed `lzma`, `xz-utils`, so I also installed `lzma-dev` to try to get headers but `cpan Compress::Raw::Lzma` still fails because it can't find `lzma.h` which should be in `lzma-dev`.

As cpan fails to install `Compress::Raw::Lzma` and that I don't want to install it manually (for maintenance purpose) as stated in the [Compress::Raw::Lzma  README](https://metacpan.org/source/PMQS/Compress-Raw-Lzma-2.082/README), I installed `libcompress-raw-lzma-perl` which is a package of `Compress::Raw::Lzma`. I have not the last version but I'm able to install `App::DubiousHTTP` with cpan.

PS : I just write this for people who may have the same problem and may see this issue in the future
